lib/actions/field-trip.js

```diff
@@ -0,0 +1,424 @@
+}
+
+// TODO: Enable saveTrip for field trip request.
+// function saveTrip (request, requestOrder) {
+//   return function (dispatch, getState) {
+//     const {callTaker, otp} = getState()
+//     const {datastoreUrl} = otp.config
+//     if (sessionIsInvalid(callTaker.session)) return
+//     const {sessionId, username} = callTaker.session
+//     const data = {
+//       sessionId: sessionId,
+//       requestId: request.id,
+//       'trip.requestOrder': requestOrder,
+//       'trip.origin': getStartOTPString(),
+//       'trip.destination': getEndOTPString(),
+//       'trip.createdBy': username,
+//       'trip.passengers': groupSize,
+//       'trip.departure': moment(groupPlan.earliestStartTime).add('hours', otp.config.timeOffset).format('YYYY-MM-DDTHH:mm:ss'),
+//       'trip.queryParams': JSON.stringify(groupPlan.queryParams)
+//     }
+//
+//     for (let i = 0; i < groupPlan.itineraries.length; i++) {
+//       const itin = groupPlan.itineraries[i]
+//       data[`itins[${i}].passengers`] = itin.groupSize
+//       data[`itins[${i}].itinData`] = otp.util.Text.lzwEncode(JSON.stringify(itin.itinData))
+//       data[`itins[${i}].timeOffset`] = otp.config.timeOffset || 0
+//
+//       const legs = itin.getTransitLegs()
+//
+//       for (let l = 0; l < legs.length; l++) {
+//         const leg = legs[l]
+//         const routeName = (leg.routeShortName !== null ? ('(' + leg.routeShortName + ') ') : '') + (leg.routeLongName || '')
+//         const tripHash = tripHashLookup[leg.tripId]
+//
+//         data[`gtfsTrips[${i}][${l}].depart`] = moment(leg.startTime).format('HH:mm:ss')
+//         data[`gtfsTrips[${i}][${l}].arrive`] = moment(leg.endTime).format('HH:mm:ss')
+//         data[`gtfsTrips[${i}][${l}].agencyAndId`] = leg.tripId
+//         data[`gtfsTrips[${i}][${l}].tripHash`] = tripHash
+//         data[`gtfsTrips[${i}][${l}].routeName`] = routeName
+//         data[`gtfsTrips[${i}][${l}].fromStopIndex`] = leg.from.stopIndex
+//         data[`gtfsTrips[${i}][${l}].toStopIndex`] = leg.to.stopIndex
+//         data[`gtfsTrips[${i}][${l}].fromStopName`] = leg.from.name
+//         data[`gtfsTrips[${i}][${l}].toStopName`] = leg.to.name
+//         data[`gtfsTrips[${i}][${l}].headsign`] = leg.headsign
+//         data[`gtfsTrips[${i}][${l}].capacity`] = itin.getModeCapacity(leg.mode)
+//         if (leg.tripBlockId) data[`gtfsTrips[${i}][${l}].blockId`] = leg.tripBlockId
+//       }
+//     }
+//     return fetch(`${datastoreUrl}/fieldtrip/newTrip`,
+//       {method: 'POST', body: data}
+//     )
+//       .then((res) => {
+//         console.log(res)
+//         if (res === -1) {
+//           alert('This plan could not be saved due to a lack of capacity on one or more vehicles. Please re-plan your trip.')
+//         } else {
+//           dispatch(fetchFieldTripDetails(request.id))
+//         }
+//       })
+//       .catch(err => {
+//         alert(`Error saving trip: ${JSON.stringify(err)}`)
+//       })
+//   }
+// }
```

Comment:
not again...

Reply:
Removed in e8cda1bd131fb1354e32d03268bb94e2ec5583cb